lib/brains/tests/test_models.py

```diff
@@ -0,0 +1,65 @@
+from braintree.payment_method_gateway import PaymentMethodGateway
+from braintree.subscription_gateway import SubscriptionGateway
+from nose.tools import eq_
+
+from lib.brains.tests.base import create_subscription, BraintreeTest
+from lib.brains.tests.test_subscription import (
+    create_method_all, successful_subscription)
+from lib.brains.tests.test_paymethod import successful_method
+
+
+class TestClose(BraintreeTest):
+    gateways = {
+        'pay': PaymentMethodGateway,
+        'sub': SubscriptionGateway,
+    }
+
+    def setUp(self):
+        super(TestClose, self).setUp()
+        self.method, self.product = create_method_all()
+        self.buyer = self.method.braintree_buyer.buyer
+        self.sub = create_subscription(self.method, self.product)
+
+    def test_no_buyer(self):
+        self.buyer.delete()
+        self.buyer.close()
+
+    def test_close(self):
+        self.mocks['pay'].delete.return_value = successful_method()
+        self.mocks['sub'].cancel.return_value = successful_subscription()
+
+        self.buyer.close()
+
+        self.mocks['pay'].delete.assert_called_with(self.method.provider_id)
+        self.mocks['sub'].cancel.assert_called_with(self.sub.provider_id)
+
+        eq_(self.method.reget().active, False)
+        eq_(self.sub.reget().active, False)
+
+    def test_listens_signal(self):
+        self.mocks['pay'].delete.return_value = successful_method()
+        self.mocks['sub'].cancel.return_value = successful_subscription()
+
+        self.buyer.close_signal.send(
+            buyer=self.buyer, sender=self.buyer.__class__)
+
+    def test_inactive_method(self):
+        # If a method is inactive, then we still go and test its payment
+        # methods.
```

Comment:
this comment doesn't make sense to me. Test its payment methods? Did you mean subscription?